### DISCUSSION TITLE
External mount binds?

### DISCUSSION THREAD

**Discussion by francoism90** [containers/podman#18220] [Q&A]
I want to keep my `docker-compose.yml` as follows:
```bash
$ cd /path/to/podman-project && cat docker-compose.yml
    volumes:
      - ./src:/src:ro,z
      - ./data:/data:ro,z
 ```
In Docker this works for me (incl. SELinux), and the container is able to access `/data`:
```bash
$ sudo mount --bind /mnt/storage/media /path/to/podman-project/data -o x-gvfs-hide
$ docker-compose up
```
I think this is because Docker runs as `root` and now I'm Podman in `rootless` mode. This results in permission denied inside the container (`system_u:object_r:unlabeled_t` vs `system_u:object_r:container_file_t`).

Any workarounds? I would prefer not to change the path in the `volumes` path, as this will be pushed to git.

Thanks!

**Comment by rhatdan** [comment]
All of the content in /mnt/storage/media  would need to be owned/writable by the user_namespace of the user account running the podman.service.  

You could remove the :z and just do.
$ sudo mount --bind /mnt/storage/media /path/to/podman-project/data -o x-gvfs-hide
$ sudo chcon -t container_file_t  /mnt/storage/media /path/to/podman-project/data 
$ docker-compose up

**Reply by rhatdan** [reply]
The other thing is once /mnt/storage/media is labeled once, it should stay labeled with container_file_t as long as you run on an SELinux system.

**Reply by francoism90** [reply]
@rhatdan Thanks! Unfortunately it doesn't work, the bind mount is empty in the container.

This is the current situation:
```
/path/to/podman-project/data // normal folder
/path/to/podman-project/data/media // bind mount
```
It can see the `/data` folder inside the container, but the `/data/media` folder is empty, really weird, as it did work before.

Do you have any idea?

**Reply by francoism90** [reply]
@rhatdan Hmm, turns out because I'm using this:
```
 x-podman:
      uidmaps:
        - "0:1:1"
        - "${UID:-1000}:0:1"
      gidmaps:
        - "0:1:1"
        - "${GID:-1000}:0:1"
```
Any other worksarounds?

**Reply by rhatdan** [reply]
Are you getting permission denied when trying to read content?

**Reply by francoism90** [reply]
@rhatdan No, just an empty folder.

I was trying different things with uid/gid mappings, and it eventually was browsable, but under the user nobody, instead of the own user.

If I map it directly, it works fine, maybe it doesn't translate a bind inside a bind? Weird, as this works fine in Docker and also without the uid/gid mappings.

**Reply by rhatdan** [reply]
I have no idea why the mount point would be empty unless the source is empty.  I don't see how the UID/GID mapping would effect this.  If you sudo the the source of the mount point is that empty?

@giuseppe Any ideas?

**Reply by francoism90** [reply]
@rhatdan The source is not empty, as it does work when not using uid/gid mapping or directly map it as volume instead. Maybe some permission to correctly mount a bind is missing inside the container?

I haven't tested with `--userns keep-id:uid=1000,gid=1000`, as I'm using `podman-compose`, resulting it cannot combine with `--pod` error.  I also didn't without `rootless` yet.

**Reply by rhatdan** [reply]
Could you get a repeater without using podman-compose? This could very well be a podman-compose error.
